### DISCUSSION TITLE
[BUG] Grant-PnPAzureADAppSitePermission throws "invalidRequest" error.

### DISCUSSION THREAD

**Discussion by dswalker1** [pnp/powershell#3040] [Q&A] [answered]
Hi All I'm running PowerShell 7.3.3 and PnP PowerShell 2.1.1. I've run Grant-PnPAzureADAppSitePermission many times in the past using an older version of PnP Powershell, but now with the 2.1.1 version I get the following error:

Grant-PnPAzureADAppSitePermission: {"error":{"code":"invalidRequest","message":"Invalid request","innerError":{"date":"2023-04-19T21:22:33","request-id":"[GUID]","client-request-id":"[GUID]"}}}

here is the command I ran (with fake id and and display name): 

 Grant-PnPAzureADAppSitePermission -AppId 0000000-0000-00000000-000000000000 -DisplayName "tst-app-reg" -Permissions Write                                                                  

To confirm that the app reg itself was not the problem I tried the command with a different app registration that I successfully configured with Grant-PnPAzureADAppSitePermission in the past and encountered the same error.

Also, I'm running this on a Windows 11 machine.

Other commands such as Connect-PnPOnline and Get-PnPAzureADAppSitePermission work fine.

Let me know if you require more information

Regards,
D. Walker

**Comment by thijsk** [comment]
I'm running into the same issue. I'm pretty sure the cmdlet worked fine yesterday, but now it does not work. My script has not changed.
```
Grant-PnPAzureADAppSitePermission: **\Sharepoint.ps1:41:15
Line |
  41 |  … ermission = Grant-PnPAzureADAppSitePermission -AppId $appId -DisplayN …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | {"error":{"code":"invalidRequest","message":"Invalid
     | request","innerError":{"date":"2023-04-21T16:40:24","request-id":"26618195-93d6-4394-a3d2-*****","client-request-id":"26618195-93d6-4394-a3d2-*****"}}}
```
     
Is there a way to see more information about the failed request?


I can still add permissions using the `m365` cli (but I want to use powershell).

**Reply by gautamdsheth** [reply, marked as answer]
Hi, there was a regression in `2.1.1` , we fixed that and it is available in the latest nightly builds. 

You can download the nightly version by doing this `Install-module PnP.PowerShell -AllowPrerelease ` or specifying the exact version like `2.1.25-nightly`

**Reply by dswalker1** [reply]
Thanks you for the information

**Reply by jrnewton** [reply]
**EDIT:**  I found my problem - I did not have Microsoft Graph / Sites.FullControl.All permission set in my app registration. 

Hi - 

I'm running 2.1.64-nightly and still getting this problem. 

Command:
Grant-PnPAzureADAppSitePermission -AppId [GUID] -DisplayName "udm-build-read" -Permissions Read

Result:
Grant-PnPAzureADAppSitePermission: {"error":{"code":"invalidRequest","message":"Invalid request","innerError":{"date":"2023-06-01T11:41:34","request-id":"[GUID]","client-request-id":"[GUID]"}}}
